openml/config.py

```diff
@@ -184,26 +189,35 @@ def _setup():
     global avoid_duplicate_runs
     global connection_n_retries
 
-    # read config file, create cache directory
-    try:
-        os.mkdir(os.path.expanduser(os.path.join("~", ".openml")))
-    except FileExistsError:
-        # For other errors, we want to propagate the error as openml does not work without cache
-        pass
+    # read config file, create openml directory
+    expanded_openml_dir = os.path.expanduser(os.path.join("~", ".openml"))
+    if not os.path.exists(expanded_openml_dir):
+        try:
+            os.mkdir(expanded_openml_dir)
+        except PermissionError:
+            warnings.warn(
```

Comment:
I would prefer using `logging` module here.

Reply:
I updated the code to use logging instead of warnings.